src/cbc_syslog/util/carbon_black_cloud.py

```diff
@@ -116,48 +116,54 @@ def build_query(cb, alert_criteria, start, end):
                       .sort_by(time_field, "ASC")
 
             # Iterate criteria options
-            for key in alert_criteria.keys():
+            for key in alert_rule.keys():
 
                 # Check for custom criteria
                 if key == "minimum_severity":
-                    query.set_minimum_severity(alert_criteria[key])
+                    query.set_minimum_severity(alert_rule[key])
 
                 # Changes with UAE v7 alerts
                 # if key == "policy_applied":
-                #     query.set_policy_applied(alert_criteria[key])
+                #     query.set_policy_applied(alert_rule[key])
 
                 # Add standard list value criteria
                 else:
-                    query.add_criteria(key, alert_criteria[key])
+                    query.add_criteria(key, alert_rule[key])
 
             return query
 
         # Iterate saved instances to fetch alerts for each instance
         for instance in self.instances:
             cb = instance["api"]
-            org_alerts = []
-
-            try:
-                # Fetch initial Alert batch
-                org_alerts.extend(build_query(cb, instance["alert_criteria"], start, end)[0:10000])
-
-                # Check if 10k limit was hit and iteratively fetch remaining alerts
-                #   by increasing start time to the last alert fetched
-                if len(org_alerts) >= 10000:
-                    last_alert = org_alerts[-1]
-                    while True:
-                        new_start = datetime.strptime(last_alert[time_field], time_format) + timedelta(milliseconds=1)
-                        overflow = build_query(cb, instance["alert_criteria"], new_start, end)
-
-                        # Extend alert list with follow up alert batches
-                        org_alerts.extend(overflow[0:10000])
-                        if len(overflow) >= 10000:
-                            last_alert = org_alerts[-1]
-                        else:
-                            break
-                alerts.extend(org_alerts)
-            except:
-                log.exception(f"Failed to fetch alerts (start: {start} - end: {end}) for org {cb.credentials.org_key}")
-                failed_orgs.append(cb.credentials.org_key)
+
+            # Perform alert fetch for each alert rule
+            for alert_rule in instance["alert_rules"]:
```

Comment:
Checking my understanding:
If there are alerts that match more than one rule then they will be exported from CBC multiple times.

I think that's fine, just need to document it that way.

Reply:
This is a great call out and will need to be documented